src/server/routes/admin/nonces.ts

```diff
@@ -0,0 +1,169 @@
+import { Static, Type } from "@sinclair/typebox";
+import { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import { eth_getTransactionCount, getRpcClient } from "thirdweb";
+import {
+  lastUsedNonceKey,
+  recycledNoncesKey,
+  sentNoncesKey,
+  splitLastUsedNonceKey,
+} from "../../../db/wallets/walletNonce";
+import { getChain } from "../../../utils/chain";
+import { normalizeAddress } from "../../../utils/primitiveTypes";
+import { redis } from "../../../utils/redis/redis";
+import { thirdwebClient } from "../../../utils/sdk";
+import { standardResponseSchema } from "../../schemas/sharedApiSchemas";
+import { walletWithAddressParamSchema } from "../../schemas/wallet";
+
+export const responseBodySchema = Type.Object({
+  result: Type.Array(
+    Type.Object({
+      walletAddress: Type.String({
+        description: "Backend Wallet Address",
+        examples: ["0xcedf3b4d8f7f1f7e0f7f0f7f0f7f0f7f0f7f0f7f"],
+      }),
+      onchainNonce: Type.String({
+        description: "Last mined nonce",
+        examples: ["0"],
+      }),
+      lastUsedNonce: Type.String({
+        description: "Last incremented nonce sent to the blockchain",
+        examples: ["0"],
+      }),
+      sentNonces: Type.Array(
+        Type.String({
+          examples: ["0"],
+        }),
+        {
+          description:
+            "Nonces that were successfully sent to the blockchain but not mined yet.",
+        },
+      ),
+      recycledNonces: Type.Array(
+        Type.String({
+          examples: ["0"],
+        }),
+        {
+          description:
+            "Nonces that were acquired but failed to be sent to the blockchain, waiting to be recycled or cancelled.",
+        },
+      ),
+      chainId: Type.Number({
+        description: "Chain ID",
+        examples: [80001],
+      }),
+    }),
+  ),
+});
+
+responseBodySchema.example = {
+  result: [
+    {
+      walletAddress: "0xcedf3b4d8f7f1f7e0f7f0f7f0f7f0f7f0f7f0f7f",
+      onchainNonce: "2",
+      lastUsedNonce: "8",
+      recycledNonces: ["6", "7"],
+      chainId: 80001,
+    },
+  ],
+};
+
+export async function getNonceDetailsRoute(fastify: FastifyInstance) {
+  fastify.route<{
+    Querystring: Static<typeof walletWithAddressParamSchema>;
+    Reply: Static<typeof responseBodySchema>;
+  }>({
+    method: "GET",
+    url: "/admin/nonces",
+    schema: {
+      summary: "Get transaction details",
+      description: "Get raw logs and details for a transaction by queueId.",
+      tags: ["Admin"],
+      operationId: "transactionDetails",
+      querystring: walletWithAddressParamSchema,
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: responseBodySchema,
+      },
+      hide: true,
+    },
+    handler: async (request, reply) => {
+      const { walletAddress, chain } = request.query;
```

Comment:
Nit in preparation for the future: We could avoid `walletWithAddressParamSchema` and make a new type that accepts the input var `chainId` instead of `chain`

Reply:
I don't like the idea of having both schemas hang around, would make things confusing... do we namespace the newer one with "v3" or similar? and later remove the namespace when we actually move to v3?